File: 1951a/main.py
def I():
    return input()


def II():
    return int(input())


def MII():
    return map(int, input().split())


def LI():
    return list(input().split())


def LII():
    return list(map(int, input().split()))

# alpha = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']

a = II()
for _ in range(a):
    b = II()
    s = I()
    ones = s.count('1')

    if ones > 2 and ones & 1 == 0:
        print("YES")
    elif ones == 0 or (ones == 2 and '11' not in s):
        print("YES")

    else:
        print("NO")

File: 1951a/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func solve(reader *bufio.Reader) {
	var b int
	var s string
	fmt.Fscan(reader, &b)
	fmt.Fscan(reader, &s)
	ones := 0
	for _, v := range s {
		if v == '1' {
			ones += 1
		}

	}

	if ones > 2 && ones&1 == 0 {
		fmt.Println("YES")
	} else if ones == 0 || (ones == 2 && !strings.Contains(s, "11")) {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}

}

func main() {
	reader := bufio.NewReader(os.Stdin)
	var a int
	fmt.Fscan(reader, &a)
	for i := 0; i < a; i++ {
		solve(reader)
	}
}
